x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapperTests.java

```diff
@@ -422,7 +434,161 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
 
     @Override
     protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
-        throw new AssumptionViolatedException("not supported");
+        // Almost like GeoShapeType but no circles
```

Comment:
Why do we skip circles? Also, shall we use GeoShapeType and skip the unsupported types in the switch?

Reply:
This field does not support circles https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-shape.html#input-structure.

Reply:
I'd still use `GeoShapeType` and map the behavior for circle to a valid one, e.g. point. That way, this won't get out of sync if another type is added to `GeoShapeType`. Mostly nit.